### PR TITLE
Istio ingress

### PR DESCRIPTION
**Pull request by tanyaveksler**

### CODE REVIEW

**Review by adisos — COMMENTED** [review]
Hi @tanyaveksler , some initial comments about the tests/examples:
- [ ] Add a simpler / basic test (style the one in the presentation)
- [ ] Current test: convert to a more realistic example (ingress controller pod from a different namespace , consider using the default one of istio which is: istio-system/istio-ingressgateway ) [current test is confusing when some pod from default namespace with a name associated with bookinfo services pods is actually an ingress controller pod, only due to its added label.]

File: tests/istio_testcases/example_policies/istio-ingress-test/virtual_services.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  gateways:
+  - default/my-gateway # can omit the namespace if gateway is in same namespace as virtual service.
+  tcp:
+  - match:
+    - port: 27017
+    route:
+    - destination:
+        host: details.prod.svc.cluster.local
+        port:
+          number: 5555
```
Comment (adisos) [inline tests/istio_testcases/example_policies/istio-ingress-test/virtual_services.yaml:55]: Should have a warning that this virtual service is ignored?

**Review by adisos — COMMENTED** [review]
Added a few more comments.

File: network-config-analyzer/IstioTrafficResourcesYamlParser.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+            if not dest:
+                self.warning(f'Missing destination in HTTPRouteDestination in the VirtualService {vs.full_name()}. '
+                             f'Ignoring http route.')
+                return
+            service = self.parse_service(dest, vs)
+            if not service:
+                self.syntax_error(f'Missing service referenced in {dest} in the VirtualService {vs.full_name()}', route)
+            target_port = None
+            port = dest.get('port')
+            if port:
```
[the remark this answers is not in the document]
Comment (tanyaveksler) [reply network-config-analyzer/IstioTrafficResourcesYamlParser.py:319]: Added a warning

File: network-config-analyzer/IstioTrafficResourcesYamlParser.py [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+        """
+        Create IngressPolicies according to the parsed Gateways and VirtualServices
+        :return list[IngressPolicy]: the resulting policies
+        """
+
+        if not self.gateways:
+            self.syntax_error('No valid Gateways found')
+        if not self.virtual_services:
+            self.syntax_error('No valid VirtualServices found')
+
```
[the remark this answers is not in the document]
Comment (tanyaveksler) [reply network-config-analyzer/IstioTrafficResourcesYamlParser.py:383]: Changed to warnings.

File: tests/istio_testcases/example_policies/istio-ingress-test/virtual_services.yaml (hunk 1 above)
Comment (tanyaveksler) [reply tests/istio_testcases/example_policies/istio-ingress-test/virtual_services.yaml:55]: Added warnings to more unsupported attributes

**Review by adisos — COMMENTED** [review]
hi @tanyaveksler , a few comments:

- [ ] change branch so its changed files is only your changes (and set of commits is only your commits)
- [ ] Merge with master -note the recent "multi layer" changes were merged to master: should be merged carefully with your changes, and fit the new flow of allowed_connections computation.
- [ ] Change "deny rules" to "allow rule" (for both k8s ingress and istio ingress)
- [ ] Fix issue of using docstring types when the the types are missing in import statements: for example, in file `IstioTrafficResourcesYamlParser.py`  , instead of writing `PeerContainer` in docstring, write `PeerContainer.PeerContainer` (since there is no import for `PeerContainer`)

**Pull request closed without merging**